content/post/087-js-ipfs-0.42.md

```diff
@@ -0,0 +1,194 @@
+
+## 🛳️ Updated Docker containers
+
+Publishing to Docker Hub has been restored so you can get a container running the latest and greatest `js-ipfs` with:
+
+```console
+$ docker pull ipfs/js-ipfs
+```
+
+You can also pull pre-releases at any time with:
+
+```console
+$ docker pull ipfs/js-ipfs:next
+```
+
+Check out all the releases at: https://hub.docker.com/r/ipfs/js-ipfs
+
+## ✨New features
+
+The [MFS][] CLI now supports nanosecond mtimes, use it with the following commands:
+
+```console
+$ jsipfs mkdir /directory --mtime 1000 --mtime-nsecs 5
+```
+
+`--mode` and `--mtime` are now also supported when piping files to `jsipfs add`:
+
+```console
+$ cat ./file.txt | jsipfs add --mode 0500
+```
+
+## 🔨 Breaking changes
+
+- [HAMT](https://en.wikipedia.org/wiki/Hash_array_mapped_trie) sharded directories are now reported as having a type of `'directory'` from `ipfs.files.stat` in line with `go-ipfs`
+- When specifying metadata and `--raw-leaves` to `ipfs.add`, if the file is small enough to fit into one block, `--raw-leaves` is ignored and a [UnixFS][] entry is created (previously the returned [CID][] would have a codec of `raw`)
+
+## 🦟 Bugs fixed
+
+- WebRTC configuration is now passed to the libp2p WebRTC transport correctly
+- Unknown CLI commands now cause the process to exit with a non-zero exit code
+- Running Prometheus metrics in Docker containers no longer prevent the container from starting
+
+## 💍 One PR to rule them all
+
+A common observation made about IPFS development is that there are so many repos to gain knowledge of, it's hard to know how everything fits together and so where a problem might originate, all of which makes it hard to start contributing.
+
+Contributors will sometimes spend precious time crafting one PR, only to be told they need to make additional PRs to repos they didn't even know existed in order to land their intial change, which can be too much of a barrier developers who have external time pressures we aren't aware of.
+
+Features must be staged accross multiple releases from multiple repos, any of which have the possibility of blocking other work streams if through human error bugs are introduced or perhaps a piece of work was not a fully specified as it needed to be.
+
+To start to address this we've consolidated `js-ipfs`, `js-ipfs-http-client` and `interface-js-ipfs-core` into [one repository](https://github.com/ipfs/js-ipfs).  This made sense because a given IPFS feature is typically implemented in `js-ipfs`, exposed over http by the http client and tests are added to the interface suite to ensure everything works together - now you will be able to make that sort of change in one PR instead of three!
+
+It's our hope that this will go some way towards simplifying IPFS development and will encourage new contributors to pitch in and help us build a the platform to power the distributed web.
```

Comment:
```suggestion
It's our hope that this will go some way towards simplifying IPFS development and will encourage new contributors to pitch in and help us build the platform to power the distributed web.
```